typing/printtyp.ml

```diff
@@ -1814,11 +1814,9 @@ let explain_escape intro ctx e =
     | None -> ignore
     | Some ctx ->  dprintf "@[%t@;<1 2>%a@]" intro type_expr ctx in
   match e with
-  | Trace.Univ Some u ->  Some(
-      dprintf "%t@,The universal variable '%s would escape its scope"
-        pre u)
-  | Trace.Univ None ->
-      Some(dprintf "%t@,An universal variable would escape its scope" pre)
+  | Trace.Univ u ->  Some(
+      dprintf "%t@,The universal variable %a would escape its scope"
+        pre type_expr u)
```

Comment:
Would it make sense to `assert` that `u` is a `Tunivar _` here, to be extra safe and enforce the implicit invariant?
Or maybe somewhere else (I haven't looked at the code related to unification traces at all).

Reply:
I am not sure if the extra-safety is needed?
The invariant is quite visibly respected at the current construction points. I imagine it could be enforced by making the trace element type private, but this seems overkill for the current 2 use cases.
Similarly, the type description should not be mutated during the error message printing, thus rechecking the invariant at this use point feels superfluous.

Reply:
Fair enough.